vlm/src/main/scala/GeoTiffRasterSource.scala

```diff
@@ -11,31 +11,9 @@ import cats.effect.IO
 import java.net.URI
 import java.nio.file.Paths
 
-import geotrellis.util.{FileRangeReader, RangeReader, StreamingByteReader}
-
-class GeoTiffRasterSource(val fileURI: String) extends RasterSource {
-  // TODO: uri on RasterReader2 should be String, because java.net.URI does not admit all valid S3 URIs
-
-  private def getByteReader(): StreamingByteReader = {
-    val javaURI = new URI(fileURI)
-    val rr =  javaURI.getScheme match {
-      case "file" | null =>
-        FileRangeReader(Paths.get(javaURI).toFile)
-
-      case "s3" => ???
-      //     val s3Uri = new AmazonS3URI(java.net.URLDecoder.decode(uri, "UTF-8"))
-      //     val s3Client = new AmazonS3Client(new AWSAmazonS3Client(new DefaultAWSCredentialsProviderChain))
-      //     S3RangeReader(s3Uri.getBucket, s3Uri.getKey, s3Client)
-
-      case scheme =>
-        throw new IllegalArgumentException(s"Unable to read scheme $scheme at $uri")
-    }
-    new StreamingByteReader(rr)
-  }
-
-  @transient private lazy val tiff: MultibandGeoTiff = GeoTiffReader.readMultiband(getByteReader, streaming = true)
-
-  def uri: URI = new URI(fileURI)
+case class GeoTiffRasterSource(uri: String) extends RasterSource {
+  @transient private lazy val tiff: MultibandGeoTiff =
```

Comment:
How would you feel about having `GeoTiffInfo` being one of the properties of `GeoTiffRasterSource`? By having that, we'd have access to additional metadata like [bandCount](https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/io/geotiff/reader/GeoTiffReader.scala#L248) which could prove useful later on.

Reply:
I'm pretty neutral about that. I would expect most of the usage of this class to actually happen through `RasterSource` interface rather than the concrete. `GeoTiffInfo` wouldn't be appropriate to expose in `RasterSource` because we expect it to abstract over GDAL. In general `GeoTiffInfo` has lots of useful metadata but I'm not sure how its going to end up being used.

Reply:
Yeah, a lot of that metadata is GeoTiff specific, so I can see it having only some limited uses. What do you think about `RasterSource` having a `bandCount` member then? We already know that's an important piece of metadata, so it'd be good to have it available as soon as possible.

Reply:
‘bandCount’ is already on ‘RasterSource’

Reply:
Not sure about the decision on VLM api, what do you mean? This mostly tries to address a direct problem with warp. I think things are going to become more clear once we make GDALRasterSource, that will need to lean on GDAL to perform the warp through virtual WRT.

Also you can see that reading band subsets will be desirable before long. This is incremental though.

Reply:
Oh, sorry. I didn't realize that `bandCount` was already on `RasterSource`. I thought we were still debating on the API of `WarpRasterSource` and whether or not it is needed? I might have misunderstood the workflowy, though